src/main/scala/centaur/test/Test.scala

```diff
@@ -107,43 +107,53 @@ object Operations {
       }
     }
 
-    new Test[Unit] {
-      def validateMetadata(workflow: SubmittedWorkflow, expectedMetadata: WorkflowMetadata, cacheHitUUID: Option[UUID] = None): Try[Unit] = {
+    new Test[WorkflowMetadata] {
+      def validateMetadata(workflow: SubmittedWorkflow, expectedMetadata: WorkflowMetadata, cacheHitUUID: Option[UUID] = None): Try[WorkflowMetadata] = {
         def checkDiff(diffs: Iterable[String]): Unit = {
           diffs match {
             case d if d.nonEmpty => throw new Exception(s"Invalid metadata response:\n -${d.mkString("\n -")}\n")
             case _ =>
           }
         }
         cleanUpImports(workflow)
-        CentaurCromwellClient.metadata(workflow).map(expectedMetadata.diff(_, workflow.id.id, cacheHitUUID)).map(checkDiff)
+
+        for {
+          actualMetadata <- CentaurCromwellClient.metadata(workflow)
+          diffs = expectedMetadata.diff(actualMetadata, workflow.id.id, cacheHitUUID)
+          _ = checkDiff(diffs)
+        } yield actualMetadata
       }
 
-      override def run: Try[Unit] = {
-        eventually(OffsetDateTime.now(), CentaurConfig.metadataConsistencyTimeout) {
-          validateMetadata(workflow, expectedMetadata, cacheHitUUID)
-        }
+      override def run: Try[WorkflowMetadata] = workflowSpec.metadataExpectation match {
+        case Some(expectedMetadata) =>
+          eventually(OffsetDateTime.now(), CentaurConfig.metadataConsistencyTimeout) {
+            validateMetadata(submittedWorkflow, expectedMetadata, cacheHitUUID)
+          }
+          // Nothing to wait for, so just return the first metadata we get back:
+        case None => CentaurCromwellClient.metadata(submittedWorkflow)
       }
     }
   }
 
   /**
     * Verify that none of the calls within the workflow are cached.
     */
-  def validateNoCacheHits(metadata: WorkflowMetadata, workflowName: String): Test[Unit] = {
+  def validateCacheHitField(metadata: WorkflowMetadata, workflowName: String, blacklistedValue: String): Test[Unit] = {
```

Comment:
"blacklistedValue" doesn't seem like the right term when looking for a cache hit